### DISCUSSION TITLE
external antenna - esp swap from the nodemcu

### DISCUSSION THREAD

**Discussion by kallejoken** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#630] [General]
i would like to exchange the esp on the nodemcu for an esp07s for the purpose of an external antenna. however, i am wondering whether the lower 6 gpios are also used. the esp07s does not have them. does anyone know?

**Comment by visualapproach** [comment]
Not too familiar with it, but depending on your pump model you need 4 or 6 (7 for audio) GPIO pins. You can cross reference the GPIO numbers with the ESP12E and figure out if it has the corresponding Dn required.

**Comment by kallejoken** [comment]
I have now soldered an external antenna directly onto the esp12f chip - but thanks
